app/assets/javascripts/views/courses.js

```diff
@@ -101,5 +111,27 @@ Yacs.views.courses = function (data) {
       c.classList[isSelected ? 'remove' : 'add']('selected');
     });
     if (isCourseSelected(c)) c.classList.add('selected');
+
+    var desc = c.querySelector('course-description');
+    var showButton = c.querySelector('show-hide-button');
+    Yacs.on('click', showButton, function (showButton, event) {
+      var descTruncated = desc.classList.contains("truncated");
+      desc.classList[descTruncated ? 'remove' : 'add']("truncated");
+      showButton.classList[descTruncated ? 'add' : 'remove']("open");
+      event.stopPropagation();
+    });
+    requireTruncation(desc, showButton);
+  });
+
+  var resizingFxn;
+  window.addEventListener("resize", function () {
```

Comment:
A third option, is to have set an interval on <courses></courses> to check if it's width has changed, and trigger the my function that is currently bound to window.onresize. This is used internally by jquery plugin to enable resize events on all elements, so there might be a sweet spot where the efficiency trade off is worth the functionality.